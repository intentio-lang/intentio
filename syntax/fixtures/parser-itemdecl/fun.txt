fun foo(a, b) {
    a;
}

fun id(x,) { x }

fun constant() { 92; }
