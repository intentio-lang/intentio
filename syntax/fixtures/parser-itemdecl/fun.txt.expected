[
  {
    "_itemDeclAnn": [],
    "_itemDeclKind": {
      "contents": {
        "_funDeclAnn": [],
        "_funDeclBody": {
          "_blockAnn": [],
          "_blockSourcePos": [
            "test.ieo",
            0,
            14
          ],
          "_blockStmts": [
            {
              "_stmtAnn": [],
              "_stmtKind": {
                "contents": {
                  "_exprAnn": [],
                  "_exprKind": {
                    "contents": {
                      "contents": {
                        "_scopeIdAnn": [],
                        "_scopeIdSourcePos": [
                          "test.ieo",
                          1,
                          4
                        ],
                        "_unScopeId": "a"
                      },
                      "tag": "ScopeId'"
                    },
                    "tag": "IdExpr"
                  },
                  "_exprSourcePos": [
                    "test.ieo",
                    1,
                    4
                  ]
                },
                "tag": "ExprStmt"
              },
              "_stmtSourcePos": [
                "test.ieo",
                1,
                4
              ]
            }
          ]
        },
        "_funDeclName": {
          "_scopeIdAnn": [],
          "_scopeIdSourcePos": [
            "test.ieo",
            0,
            4
          ],
          "_unScopeId": "foo"
        },
        "_funDeclParams": [
          {
            "_scopeIdAnn": [],
            "_scopeIdSourcePos": [
              "test.ieo",
              0,
              8
            ],
            "_unScopeId": "a"
          },
          {
            "_scopeIdAnn": [],
            "_scopeIdSourcePos": [
              "test.ieo",
              0,
              11
            ],
            "_unScopeId": "b"
          }
        ],
        "_funDeclSourcePos": [
          "test.ieo",
          0,
          0
        ]
      },
      "tag": "FunItemDecl"
    },
    "_itemDeclSourcePos": [
      "test.ieo",
      0,
      0
    ]
  },
  {
    "_itemDeclAnn": [],
    "_itemDeclKind": {
      "contents": {
        "_funDeclAnn": [],
        "_funDeclBody": {
          "_blockAnn": [],
          "_blockSourcePos": [
            "test.ieo",
            4,
            11
          ],
          "_blockStmts": [
            {
              "_stmtAnn": [],
              "_stmtKind": {
                "contents": {
                  "_exprAnn": [],
                  "_exprKind": {
                    "contents": {
                      "contents": {
                        "_scopeIdAnn": [],
                        "_scopeIdSourcePos": [
                          "test.ieo",
                          4,
                          13
                        ],
                        "_unScopeId": "x"
                      },
                      "tag": "ScopeId'"
                    },
                    "tag": "IdExpr"
                  },
                  "_exprSourcePos": [
                    "test.ieo",
                    4,
                    13
                  ]
                },
                "tag": "ExprStmt"
              },
              "_stmtSourcePos": [
                "test.ieo",
                4,
                13
              ]
            }
          ]
        },
        "_funDeclName": {
          "_scopeIdAnn": [],
          "_scopeIdSourcePos": [
            "test.ieo",
            4,
            4
          ],
          "_unScopeId": "id"
        },
        "_funDeclParams": [
          {
            "_scopeIdAnn": [],
            "_scopeIdSourcePos": [
              "test.ieo",
              4,
              7
            ],
            "_unScopeId": "x"
          }
        ],
        "_funDeclSourcePos": [
          "test.ieo",
          4,
          0
        ]
      },
      "tag": "FunItemDecl"
    },
    "_itemDeclSourcePos": [
      "test.ieo",
      4,
      0
    ]
  },
  {
    "_itemDeclAnn": [],
    "_itemDeclKind": {
      "contents": {
        "_funDeclAnn": [],
        "_funDeclBody": {
          "_blockAnn": [],
          "_blockSourcePos": [
            "test.ieo",
            6,
            15
          ],
          "_blockStmts": [
            {
              "_stmtAnn": [],
              "_stmtKind": {
                "contents": {
                  "_exprAnn": [],
                  "_exprKind": {
                    "contents": {
                      "_litAnn": [],
                      "_litKind": {
                        "contents": 92,
                        "tag": "IntegerLit"
                      },
                      "_litSourcePos": [
                        "test.ieo",
                        6,
                        17
                      ],
                      "_litText": "92"
                    },
                    "tag": "LitExpr"
                  },
                  "_exprSourcePos": [
                    "test.ieo",
                    6,
                    17
                  ]
                },
                "tag": "ExprStmt"
              },
              "_stmtSourcePos": [
                "test.ieo",
                6,
                17
              ]
            }
          ]
        },
        "_funDeclName": {
          "_scopeIdAnn": [],
          "_scopeIdSourcePos": [
            "test.ieo",
            6,
            4
          ],
          "_unScopeId": "constant"
        },
        "_funDeclParams": [],
        "_funDeclSourcePos": [
          "test.ieo",
          6,
          0
        ]
      },
      "tag": "FunItemDecl"
    },
    "_itemDeclSourcePos": [
      "test.ieo",
      6,
      0
    ]
  }
]
